tools/deploy-cli/deploy-cli.go

```diff
@@ -0,0 +1,324 @@
+		return err
+	}
+
+	return deployWithKustomizeAndApply(tempIronicOverlay)
+}
+
+// deployBMO generates the YAML for the Bare Metal Operator using Kustomize
+// and applies it to the Kubernetes cluster.
+func deployBMO(data *DeployContext, tempBMOOverlay string) error {
+
+	inputFile := "templates/bmo-kustomize.tpl"
+	kustomizeFile := filepath.Join(tempBMOOverlay, "kustomization.yaml")
+	if err := RenderEmbedTemplateToFile(data.TemplateFiles, inputFile, kustomizeFile, data); err != nil {
+		return err
+	}
+	ironicEnvSrc := "https://raw.githubusercontent.com/metal3-io/baremetal-operator/main/config/default/ironic.env"
+	ironicEnvDst := filepath.Join(tempBMOOverlay, "ironic.env")
+	if err := DownloadFile(ironicEnvSrc, ironicEnvDst); err != nil {
+		return err
+	}
+
+	return deployWithKustomizeAndApply(tempBMOOverlay)
+}
+
+// getKubeconfigPath determines the kubeconfig path from KUBECTL_ARGS, KUBECONFIG_PATH, or defaults to ~/.kube/config.
+func getKubeconfigPath() (string, error) {
+	kubectlArgs := os.Getenv("KUBECTL_ARGS")
+	kubeconfigPrefix := "--kubeconfig="
+	regexPattern := regexp.MustCompile(`^--kubeconfig=[\w/.-]+$`)
+	var kubeconfigPath string
+
+	if strings.Contains(kubectlArgs, kubeconfigPrefix) && regexPattern.MatchString(kubectlArgs) {
+		kubeconfigPath = strings.TrimPrefix(kubectlArgs, kubeconfigPrefix)
+	} else if kubectlArgs != "" {
+		return "", fmt.Errorf("error: invalid format in KUBECTL_ARGS. Expected format: '--kubeconfig=/path/to/kubeconfig'")
+	}
+
+	if kubeconfigPath == "" {
+		kubeconfigPath = os.Getenv("KUBECONFIG_PATH")
+		if kubeconfigPath == "" {
+			homeDir, err := os.UserHomeDir()
+			if err != nil {
+				return "", fmt.Errorf("failed to get user home directory: %v", err)
+			}
+			kubeconfigPath = filepath.Join(homeDir, ".kube", "config")
+		}
+	}
+
+	// Verify the file exists and is readable
+	_, err := os.Stat(kubeconfigPath)
+	if os.IsNotExist(err) || err != nil {
+		return "", fmt.Errorf("specified kubeconfig file does not exist or is not readable: %s", kubeconfigPath)
+	}
+
+	return kubeconfigPath, nil
+}
+
+// generateClusterProxy creates a new ClusterProxy instance using the provided kubeconfig path.
+func generateClusterProxy(kubeconfigPath string) (*framework.ClusterProxy, error) {
+	scheme := runtime.NewScheme()
+	framework.TryAddDefaultSchemes(scheme)
+	clusterProxy := framework.NewClusterProxy("deploy-cli", kubeconfigPath, scheme)
+	if clusterProxy == nil {
+		return nil, fmt.Errorf("failed to create cluster proxy")
+	}
+	return &clusterProxy, nil
+}
+
+// deployWithKustomizeAndApply first generates the YAML configuration by running Kustomize build on the overlay directory,
+// then outputs the generated YAML to a temp file within the overlay dir, and then applies
+// the configuration to the Kubernetes cluster specified in the kubeconfig file.
+func deployWithKustomizeAndApply(overlayPath string) error {
```

Comment:
This also feels like something very close to what we have in e2e already. Reuse if possible, or explain why it is not.

Reply:
Thanks. I'll do that